simulation/traffic_simulator/src/data_type/lanelet_pose.cpp

```diff
@@ -76,6 +80,23 @@ auto CanonicalizedLaneletPose::canonicalize(
       ") is invalid, please check lanelet length, connection and entity route.");
   }
 }
+
+auto CanonicalizedLaneletPose::getAlternativeLaneletPoseBaseOnShortestRouteFrom(
+  LaneletPose from, const std::shared_ptr<hdmap_utils::HdMapUtils> & hdmap_utils) const
+  -> LaneletPose
+{
+  std::vector<std::int64_t> shortest_route =
+    hdmap_utils->getRoute(from.lanelet_id, lanelet_poses_[0].lanelet_id);
```

Comment:
Please care when the `lanelet_poses_.empty() == true`

Reply:
I've added 
```
 if (lanelet_poses_.empty()) {
    return std::nullopt;
  }
  ```